src/qmapshack/mouse/line/IMouseEditLine.cpp

```diff
@@ -350,6 +353,38 @@ void IMouseEditLine::slotTrackRouting()
     canvas->reportStatus("Routino", QString());
 }
 
+void IMouseEditLine::slotOptimize()
+{
+    canvas->reportStatus(key.item, "");
+    canvas->reportStatus("Optimization", tr("<b>Started Optimization.</b><br/>"));
+
+    int response = 0;
+    try
+    {
+        response = optimizer.optimize(points);
+    }
+    catch(const QString &msg)
+    {
+        response = -1;
+        lineOp->showRoutingErrorMessage(msg);
+    }
+
+    if(response == -1)
+    {
+        canvas->reportStatus("Optimization", tr("<b>Optimization failed.</b><br/><b>Note:</b> The selected router must be able to route on-the-fly. Offline routers usually can do, online routers can't.<br/>"));
```

Comment:
For the sanity of the translators we should avoid html tags in the translation macros. Makes their life much easier. Have a look at `CMouseRangeTrk.cpp` line 45 ff. 

Please adjust all other strings, too.